.github/workflows/lint.yml

```diff
@@ -0,0 +1,22 @@
+name: lint
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  lint:
```

Comment:
why not add this as a job in the `jobs` list of `main.yml`?

Reply:
This job needs to run on pull request. But existing job needs to run on master push.
So we need separate these jobs into other workflows.